### PR TITLE
DCAT RDF generation

### PR DESCRIPTION
**Pull request by amercader**
Placeholder issue for CKAN -> DCAT

### CODE REVIEW

**Comment by amercader** [comment]
Progress:

- [ ] Extend RDFParser and RDFProfile to generate graphs and serialize datasets / catalog
- [ ] Dataset endpoint
- [ ] Catalog endpoint
- [ ] Org endpoint
- [ ] Handle URIs: create suitable defaults, config option for base URI
- [ ] Better resource format support
- [ ] Spatial field on datasets
- [ ] JSON-LD support
- [ ] Docs
- [ ] Tests

If enough time
- [ ] Content negotiation

**Comment by amercader** [comment]
@davidread @Hoedic This is still WIP but already functional. If you are interested feel free to have a play and let me know your thoughts

File: ckanext/dcat/processors.py [hunk 1]
```diff
@@ -146,13 +152,123 @@ def datasets(self):
...
+
+    Supports different profiles which are the ones that will generate
+    the RDF graph.
+    '''
+
+    def graph_from_dataset(self, dataset_dict):
+        '''
+        Given a CKAN dataset dict, creates a graph using the loaded profiles
+
+        The class RDFLib graph (accessible via `parser.g`) will be updated by
```
Comment (davidread) [inline ckanext/dcat/processors.py]: serializer.g instead of parser.g?

Comment (amercader) [inline ckanext/dcat/processors.py]: you are right, forgot to update the docstrings on b2887d79b28bad634172b3f4ee657f65f793b3c2

File: ckanext/dcat/utils.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    Returns an URI for the dataset
+
+    This will be used to uniquely reference the dataset on the RDF
+    serializations.
+
+    The value will be the first found of:
+
+        1. The value of the `uri` field
+        2. The value of an extra with key `uri`
+        3. `catalog_uri()` + '/dataset/' + `id` field
```
Comment (davidread) [inline ckanext/dcat/utils.py]: I would have thought package.name was preferable to package.id. I realise that the name might change and "cool URIs don't change". But it's pretty rare. And Linked Data people really like readable URIs, and my feeling is that that is more important. I'd love to get an 'expert' opinion on this though.

**Comment by davidread** [comment]
Aside from the minor comments I've left to ponder, this looks great. Loving the ability to customize the profile.

I'm not sure we would use yet though, until we can cache the results somewhere. I wonder if we shouldn't just stick the serializations in a table in the database. Or SOLR is pretty cache-like - we cache the package dict - but I can it's not ideal to include this in the package object too as it slows up the indexing of the catalogue further, and the dcat is only of secondary importance. Redis would mean it's kept in memory all the time which is unnecessary. So I'm favouring database.

File: ckanext/dcat/utils.py (hunk 2 above)
Comment (amercader) [inline ckanext/dcat/utils.py]: @davidread I can see where you are coming from, I just feel that specially when dealing when harvested datasets coming from different sources you really want that to be immutable. +1 on getting the opinion of an expert if we can find one :)

**Pull request merged**